ai2thor/controller.py

```diff
@@ -481,19 +482,44 @@ def __init__(
                     DeprecationWarning,
                 )
 
+            # Let's set the scene for them!
+            if scene is None:
+                scenes_in_build = self.scenes_in_build
+                if not scenes_in_build:
+                    raise RuntimeError("No scenes are in your build of AI2-THOR!")
+
+                # we use FloorPlan28 because it is used for all the testing actions
+                if "FloorPlan28" in scenes_in_build:
+                    scene = "FloorPlan28"
+                else:
+                    # use a robothor scene
+                    robothor_scenes = set(self.robothor_scenes())
+
+                    # prioritize robothor if locobot is being used
+                    robothor_scenes_in_build = robothor_scenes.intersection(scenes_in_build)
+
+                    if (
+                        "locobot" == unity_initialization_parameters.get("agentMode", "default").lower() and
+                        robothor_scenes_in_build
+                    ):
+                        scene = random.choice(list(robothor_scenes_in_build))
+                    else:
+                        ithor_scenes = set(self.ithor_scenes())
+                        ithor_scenes_in_build = ithor_scenes.intersection(scenes_in_build)
+                        if ithor_scenes_in_build:
+                            # Prioritize iTHOR because that's what the default agent best uses.
+                            scene = random.choice(list(ithor_scenes_in_build))
+                        else:
+                            # perhaps only using RoboTHOR or using only custom scenes
+                            scene = random.choice(list(scenes_in_build))
```

Comment:
I would argue strongly for this logic to change. It should be something like:
1. If using "locobot" (or "bot"!) then use the first element of `list(sorted(robothor_scenes_in_build))`.
2. Otherwise use the first element of `list(sorted(ithor_scenes_in_build))`.
 
There is no reason to introduce any randomness or favor scene 28 (especially as 28 is often used for testing!).

Reply:
Fixed.

Just a small typo, but since `ithor_scenes_in_build` and `robothor_scenes_in_build` are sets, they had to be convert to a list first before sorting, so `list(sorted(ithor_scenes_in_build))` is `sorted(list(ithor_scenes_in_build))`.

Reply:
I originally went with random scenes to show off the diversity, but fair point on that being unexpected.